example/src/App.tsx

```diff
@@ -65,35 +62,50 @@ const RenderListItem = React.memo(
     const styles = stylesheet({ currentUser: item.fromCurrentUser });
     const [isLoading, setIsLoading] = useState(true);
 
-    const handleButtonAction = () => {
-      if (playerState === PlayerState.stopped) {
-        setCurrentPlaying(item.path);
-      } else if (playerState === PlayerState.playing) {
-        ref.current?.pausePlayer();
+    const handlePlayPauseAction = async () => {
+      let currentPlayer = currentPlayingRef?.current;
+
+      // If no player or if current player is stopped just start it!
+      if (
+        currentPlayer == null ||
+        currentPlayer.currentState === PlayerState.stopped
+      ) {
+        currentPlayingRef = ref;
+        await currentPlayingRef.current?.startPlayer({
+          finishMode: FinishMode.stop,
+        });
       } else {
-        ref.current?.resumePlayer();
+        // If we are recording do nothing
+        if (currentPlayer.currentState === RecorderState.recording) {
+          return;
+        }
+
+        // Pause current player if it was playing
+        if (currentPlayer.currentState === PlayerState.playing) {
+          await currentPlayingRef?.current?.pausePlayer();
+        }
+
```

Comment:
```suggestion
  if(currentPlayer.currentState === PlayerState.paused && currentPlayer.playerKey() === ref?.current?.playerKey()) {
          await currentPlayingRef?.current?.resumePlayer();
        }
```

If player is paused and user is tapping on resume button then it should resume again.

Reply:
Yes, I got tired yesterday and was not thinking straight; I should have put the PR in the draft since I had yet to finish. I just got it this morning with some hours of sleep

Reply:
Just pushed my latest changes, it is now working correctly and resuming if it was already previously paused